lib/monitor.sh

```diff
@@ -0,0 +1,47 @@
+
+monitor_memory_usage() {
+  local output_file=$1
+  local command="${@:2}"
+
+  # Run the command in the background
+  exec $command &
```

Comment:
I am not certain about the quoting implications here, I think this may break with spaces in arguments.

I would do the following:

```
local output_file="$1"
shift # drop $1 and leave other arguments in place but with new numbering
exec "$@" &
```

Reply:
you may have to "failsafe" the `$@` access via `${@:-}`